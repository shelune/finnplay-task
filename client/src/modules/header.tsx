import React, { FC, useCallback } from "react";

import { request } from "../utils/client";
import css from "./header.module.scss";
import Logo from "../assets/images/logo.png";
import { ReactComponent as ProfileIcon } from "../assets/icons/profile.svg";

type Props = {
  setUser: (user: { username: string } | null) => void;
  username: string;
};

export const Header: FC<Props> = ({ setUser, username }) => {
  const logout = useCallback(async () => {
    await request("logout", "DELETE", {}, {});
  }, []);

  return (
    <div className={css.header}>
      <div className={css.headerContainer}>
        <div className={css.logo}>
          <img src={Logo} alt="logo" />
        </div>
        <div className={css.user}>
          <div className={css.username}>{username}</div>
          <div className={css.logout}>
            <ProfileIcon className={css.logoutIcon} />
            <a
              href="#0"
              className={css.logoutLink}
              onClick={(e) => {
                e.preventDefault();
                // eslint-disable-next-line @typescript-eslint/no-floating-promises
                logout();
                setUser(null);
              }}
            >
              Logout
            </a>
          </div>
        </div>
      </div>
    </div>
  );
};
