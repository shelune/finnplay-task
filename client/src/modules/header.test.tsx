import React from "react";
import { act, render } from "@testing-library/react";
import { Header } from "./header";

describe("Header", () => {
  it("should render", () => {
    const { container } = render(
      <Header username="player1" setUser={jest.fn()} />,
    );
    expect(container).toMatchSnapshot();
  });

  it("should call logout when clicking on logout", () => {
    const mockSetUser = jest.fn();
    const { getByText } = render(
      <Header username="player1" setUser={mockSetUser} />,
    );
    act(() => {
      getByText("Logout").click();
    });
    expect(mockSetUser).toHaveBeenCalled();
  });
});
