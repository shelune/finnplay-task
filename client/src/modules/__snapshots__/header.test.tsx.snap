// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render 1`] = `
<div>
  <div
    class="header"
  >
    <div
      class="headerContainer"
    >
      <div
        class="logo"
      >
        <img
          alt="logo"
          src="logo.png"
        />
      </div>
      <div
        class="user"
      >
        <div
          class="username"
        >
          player1
        </div>
        <div
          class="logout"
        >
          <svg
            class="logoutIcon"
          >
            profile.svg
          </svg>
          <a
            class="logoutLink"
            href="#0"
          >
            Logout
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
